backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+
+        const timestamp = new Date(Date.now());
+        await pool.query('insert into games (user1, user2, dimension, state, created_at) values ($1, $2, $3, $4, $5) returning *', [req.user.id, check.rows[0].id, dimension, 'invited', timestamp]);
+        return res.status(201).json({msg: 'Game created'});
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+
+// Get all user games
+gameRouter.get('/games', async (req, res) => {
+    let activeGames = [];
+    let gameInvitations = [];
+
+    try {
+        const userStatus = await pool.query('select wins, loses from users where id = $1', [req.user.id]);
+        if (userStatus.rows.length === 0) {
+            return res.status(400).json({msg: 'User does not exists'});
+        }
+
+        const gamesShots = await pool.query(`select g.id as game_id, u.nickname as opponent, dimension as board_dimension, sum(case when user_id = $1 then 1 else 0 end) as hits, sum(case when user_id = $1 then 0 else 1 end) as bombed 
+        from games g join shots s on g.id = s.game_id join users u on (u.id = g.user1 or u.id = g.user2) and u.id <> $1 where (user1 = $1 or user2 = $1) and s.hit = true and (g.state = $2 or g.state = $3) group by 1,2,3`,
+        [req.user.id, 'user1_turn', 'user2_turn']);
+        if (gamesShots.rows.length > 0) {
+            activeGames.push(...gamesShots.rows);
+        }
+
+        const otherGames = await pool.query(`select g.id, g.user1, g.user2, u.nickname as opponent, g.dimension, state
+        from games g join users u on (u.id = g.user1 or u.id = g.user2) and u.id <> $1 where (user1 = $1 or user2 = $1) and (state = $2 or state = $3 or state = $4 or state = $5) order by g.created_at`,
+        [req.user.id, 'invited', 'accepted', 'user1_ready', 'user2_ready']);
+
+        otherGames.rows.forEach(game => {
+            if (game.state === 'invited') {
+                gameInvitations.push({
+                    game_id: game.id,
+                    opponent: game.opponent,
+                    board_dimension: game.dimension
+                });
+            }
+        });
+
+        otherGames.rows.forEach(game => {
+            if (game.state !== 'invited') {
+                activeGames.push({
+                    game_id: game.id,
+                    opponent: game.opponent,
+                    board_dimension: game.dimension,
+                    hits: 0,
+                    bombed: 0
+                });
+            }
+        });
+
+        const finalResults = {
+            user_score: {
+                wins: userStatus.rows[0].wins,
+                loses: userStatus.rows[0].loses
+            },
+            invitations: gameInvitations,
+            active_games: activeGames
+        }
+
+        return res.status(200).json(finalResults);
+    } catch (e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+
+// Update game- accept state
+gameRouter.put('/games/:game_id', async (req, res) => {
+    try {
```

Comment:
if you want, you can create a middleware function:
```
const verifyAcceptDecilne = async (req, res, next) => { ... }
```
with lines 145 - 154, and use it both here and in the `delete` API to avoid code duplication, e.g.:
```
gameRouter.put('/games/:game_id', verifyAcceptDecilne , async (req, res) => {...}
```
and similarly for the `delete` API.